torch/csrc/jit/python/script_init.cpp

```diff
@@ -866,6 +866,53 @@ void initJitScriptBindings(PyObject* module) {
                 // Similar to Tensor's `__hash__`, which is `id()`.
                 return std::hash<c10::ivalue::Object*>{}(self._ivalue().get());
               })
+          .def(
```

Comment:
Re-using the pickling infra for torch bind objects looks good to me since all existing usage of script objects supports serde. Can we somehow re-use the two functions defined for pickling to reduce some redundant code?